programs/account-compression/src/processor/initialize_address_merkle_tree.rs

```diff
@@ -31,21 +30,12 @@ pub fn process_initialize_address_merkle_tree(
 ) -> Result<()> {
     let mut address_merkle_tree = address_merkle_tree_loader.load_init()?;
 
-    address_merkle_tree.index = index;
-    address_merkle_tree.owner = owner;
-    address_merkle_tree.delegate = delegate.unwrap_or_default();
-    address_merkle_tree.tip = tip;
-    address_merkle_tree.associated_queue = associated_queue;
-
-    address_merkle_tree.rollover_fee = match rollover_threshold {
-        Some(rollover_threshold) => {
-            compute_rollover_fee(rollover_threshold, height, rent).map_err(ProgramError::from)?
-        }
-        None => 0,
-    };
-    address_merkle_tree.rollover_threshold = rollover_threshold.unwrap_or(u64::MAX);
-    address_merkle_tree.rolledover_slot = u64::MAX;
-    address_merkle_tree.close_threshold = close_threshold.unwrap_or(u64::MAX);
+    address_merkle_tree.init(
+        AccessMetadata::new(owner, delegate),
+        RolloverMetadata::new(rent, index, rollover_threshold, tip)?,
```

Comment:
btw close_threshold could be part of the RolloverMetaData.
- the purpose of the close threshold is to mark how long the period between Merkle tree rollover and closing the account should be, right now it's a placeholder we will add this in the future.